pkg/question/question_test.go

```diff
@@ -1139,3 +1185,400 @@ func cleanupQuestionTest() string {
+			},
+			{
+				VpcId:     aws.String("vpc-67890"),
+				CidrBlock: aws.String("some block"),
+				IsDefault: aws.Bool(false),
+			},
+			{
+				VpcId:     aws.String(defaultVpc),
+				CidrBlock: aws.String("some block"),
+				IsDefault: aws.Bool(false),
+			},
+			{
+				VpcId:     aws.String("vpc-41239"),
+				CidrBlock: aws.String("some block"),
+				IsDefault: aws.Bool(false),
+			},
+		},
+	}
+
+	answer, err := question.AskVpc(testEC2, defaultVpc)
+	th.Ok(t, err)
+	th.Equals(t, defaultVpc, *answer)
+
+	cleanupQuestionTest()
+}
+
+func TestAskSubnet_WithDefault(t *testing.T) {
+	const testVpc = "vpc-12345"
+	const defaultSubnet = "subnet-12345"
+	initQuestionTest(t, "\n")
+
+	testEC2.Svc = &th.MockedEC2Svc{
+		Subnets: []*ec2.Subnet{
+			{
+				SubnetId:         aws.String("subnet-01894"),
+				VpcId:            aws.String(testVpc),
+				CidrBlock:        aws.String("some block"),
+				AvailabilityZone: aws.String("some az"),
+				Tags: []*ec2.Tag{
+					{
+						Key:   aws.String("Name"),
+						Value: aws.String("test subnet"),
+					},
+				},
+			},
+			{
+				SubnetId:         aws.String("subnet-67890"),
+				VpcId:            aws.String(testVpc),
+				CidrBlock:        aws.String("some block"),
+				AvailabilityZone: aws.String("some az"),
+			},
+			{
+				SubnetId:         aws.String("subnet-77245"),
+				VpcId:            aws.String(testVpc),
+				CidrBlock:        aws.String("some block"),
+				AvailabilityZone: aws.String("some other az"),
+			},
+			{
+				SubnetId:         aws.String(defaultSubnet),
+				VpcId:            aws.String(testVpc),
+				CidrBlock:        aws.String("some block"),
+				AvailabilityZone: aws.String("some az"),
+			},
+		},
+	}
+
+	answer, err := question.AskSubnet(testEC2, testVpc, defaultSubnet)
+	th.Ok(t, err)
+	th.Equals(t, defaultSubnet, *answer)
+
+	cleanupQuestionTest()
+}
+
+func TestAskSubnetPlaceholder_WithDefault(t *testing.T) {
+	const defaultAz = "us-east-2"
+	initQuestionTest(t, "\n")
+
+	testEC2.Svc = &th.MockedEC2Svc{
+		AvailabilityZones: []*ec2.AvailabilityZone{
+			{
+				ZoneName: aws.String("us-east-1"),
+				ZoneId:   aws.String("some id"),
+			},
+			{
+				ZoneName: aws.String(defaultAz),
+				ZoneId:   aws.String("some id"),
+			},
+			{
+				ZoneName: aws.String("us-west-1"),
+				ZoneId:   aws.String("some id"),
+			},
+			{
+				ZoneName: aws.String("us-west-2"),
+				ZoneId:   aws.String("some id"),
+			},
+		},
+	}
+
+	answer, err := question.AskSubnetPlaceholder(testEC2, defaultAz)
+	th.Ok(t, err)
+	th.Equals(t, defaultAz, *answer)
+
+	cleanupQuestionTest()
+}
+
+func TestAskBootScriptConfirmation_WithDefault(t *testing.T) {
+	defaultBootScript := ""
+	defaultConfirmation := cli.ResponseNo
+	initQuestionTest(t, "\n")
+
+	confirmation := question.AskBootScriptConfirmation(testEC2, defaultBootScript)
+	th.Equals(t, confirmation, defaultConfirmation)
+
+	cleanupQuestionTest()
+}
+
+func TestAskBootScript_WithDefault(t *testing.T) {
+	defaultBootScript := ""
```

Comment:
We have [tests for `ValidateFilepath()`](https://github.com/GavinBurris42/aws-simple-ec2-cli/blob/main/pkg/ec2helper/ec2helper_test.go#L1366-L1374) that create a temporary file. Could you use a similar mechanism to test this function? Empty string is the initial default anyway, so if possible I'd like to have this test exercise the override mechanism, otherwise this test passing doesn't give us much information.